src/Streamly/Internal/Data/Unfold/Types.hs

```diff
@@ -63,3 +67,86 @@ map f (Unfold ustep uinject) = Unfold step uinject
 instance Functor m => Functor (Unfold m a) where
     {-# INLINE fmap #-}
     fmap = map
+
+------------------------------------------------------------------------------
+-- Applicative
+------------------------------------------------------------------------------
+
+{-# INLINE const #-}
+const :: Applicative m => m b -> Unfold m a b
+const m = Unfold step inject
+
+    where
+
+    inject _ = pure False
+
+    step False = (`Yield` True) <$> m
+    step True = pure Stop
+
+-- | Outer product discarding the first element.
+--
+-- /Unimplemented/
+--
+{-# INLINE_NORMAL apSequence #-}
+apSequence :: -- Monad m =>
+    Unfold m a b -> Unfold m a c -> Unfold m a c
+apSequence (Unfold _step1 _inject1) (Unfold _step2 _inject2) = undefined
+
+-- | Outer product discarding the second element.
+--
+-- /Unimplemented/
+--
+{-# INLINE_NORMAL apDiscardSnd #-}
+apDiscardSnd :: -- Monad m =>
+    Unfold m a b -> Unfold m a c -> Unfold m a b
+apDiscardSnd (Unfold _step1 _inject1) (Unfold _step2 _inject2) = undefined
+
+data Cross a s1 b s2 = CrossOuter a s1 | CrossInner a s1 b s2
+
+-- | Create a cross product (vector product or cartesian product) of the
+-- output streams of two unfolds.
+--
+{-# INLINE_NORMAL cross #-}
+cross :: Monad m => Unfold m a b -> Unfold m a c -> Unfold m a (b, c)
+cross (Unfold step1 inject1) (Unfold step2 inject2) = Unfold step inject
+
+    where
+
+    inject a = do
```

Comment:
I like `seed` here as it is shared and is the same in the entire scope of the function.